ethcore/builtin/src/lib.rs

```diff
@@ -211,89 +227,128 @@ impl Builtin {
 
 	/// Whether the builtin is activated at the given block number.
 	pub fn is_active(&self, at: u64) -> bool {
-		at >= self.activate_at
+		self.pricer.keys().rev().any(|&other_at| at >= other_at)
 	}
 }
 
 impl From<ethjson::spec::Builtin> for Builtin {
 	fn from(b: ethjson::spec::Builtin) -> Self {
-		let pricer: Box<dyn Pricer> = match b.pricing {
-			ethjson::spec::Pricing::Blake2F { gas_per_round } => {
-				Box::new(gas_per_round)
+		// TODO(niklasad1): change this to `try_from` and propogate the error accordingly
+		let native = EthereumBuiltin::from_str(&b.name).unwrap();
+		let pricer = match b.pricing {
+			ethjson::spec::builtin::Pricing::Single(pricer) => {
+				// TODO(niklasad1): change chain specs and don't enable by default!?
+				// Seems more reasonable to do it the otherway around but would be a breaking change
+				map![b.activate_at.map_or(0, Into::into) => pricer.into()]
```

Comment:
So if `activate_at` is missing in the JSON, it's on? Do I read that right? Seems a bit dangerous, but if that's the current default we should probably not change it.

Reply:
Yes, that is correct and I agree.